userspace/plugin/plugin_api.h

```diff
@@ -970,13 +1031,40 @@ typedef struct
 	//
 	// Return an updated set of metrics provided by this plugin.
 	// Required: no
+	// Arguments:
 	// - s: the plugin state, returned by init(). Can be NULL.
 	// - num_metrics: lenght of the returned metrics array.
 	//
 	// Return value: Pointer to the first element of the metrics array.
 	// 'num_metrics' must be set to the lenght of the array before returning
 	// and it can be set to 0 if no metrics are provided.
 	ss_plugin_metric* (*get_metrics)(ss_plugin_t* s, uint32_t* num_metrics);
+
+	//Capture listening capability API
+	struct
+	{
+		//
+		// Called by the framework when the event capture opens.
+		//
+		// Required: no
+		// Arguments:
+		// - s: the plugin state, returned by init(). Can be NULL.
```

Comment:
Hmm if a plugin failed to initialize, it failed. I don't think we should call any further of its methods, except to clean it up.

Reply:
If I'm not wrong a plugin can return a NULL state pointer without failing.
This comment just wants to suggest the user to check if the state pointer is not null before dereferencing it.

Reply:
Hmm okay then :)